dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/advanced/ConsentAwareFileOrchestrator.kt

```diff
@@ -46,6 +46,10 @@ internal open class ConsentAwareFileOrchestrator(
         return null
     }
 
+    override fun getAllReadableFiles(): List<File> {
+        return grantedOrchestrator.getAllFiles()
```

Comment:
just curious: can we by any chance get a `writable` file here? because `getAllFiles` says it is both types.

Reply:
normally when calling this method we should have stopped adding any tasks into the persistence executor, that is the reason why we are draining it before calling the flush method. The flush method will take all the files (including the writable one) and send them to make sure we have sent all the event before stoping the SDK. In our nightly tests we are calling this method in the test rule in the `After` method after the process was idled so it should not be the case for a broken payload. For now I did not experience any issues but I guess the time will tell and we will adapt this method by adding more locks and checks but for now I'd say KISS